web/docs/tql2/operators/metrics.md

```diff
@@ -46,12 +46,56 @@ Contains information about all accessed API endpoints, emitted once per second.
 The schema of the record `params` depends on the API endpoint used. Refer to the
 [API documentation](/api) to see the available parameters per endpoint.
 
+### `tenzir.metrics.caf`
+
+Contains metrics about the CAF (C++ Actor Framework) runtime system.
+
+:::warning Aimed at Developers
+CAF metrics primarily exist for debugging purposes. Actor names and other
+details contained in these metrics are documented only in source code, and we
+may change them without notice. Do not rely on specific actor names or metrics
+in production systems.
+:::
+
+| Field       | Type           | Description                               |
+| :---------- | :------------- | :---------------------------------------- |
+| `system`    | `record`       | Metrics about the CAF actor system.       |
+| `middleman` | `record`       | Metrics about CAF's network layer.        |
+| `actors`    | `list<record>` | Per-actor metrics for all running actors. |
+
+The record `system` has the following schema:
+
+| Field                | Type    | Description                                     |
+| :------------------- | :------ | :---------------------------------------------- |
+| `running_actors`     | `int64` | Number of currently running actors.             |
+| `queued_messages`    | `int64` | Number of messages waiting in actor mailboxes.  |
+| `processed_messages` | `int64` | Number of messages processed since last metric. |
+| `rejected_messages`  | `int64` | Number of messages rejected since last metric.  |
+
+The record `middleman` has the following schema:
+
+| Field                    | Type       | Description                                           |
+| :----------------------- | :--------- | :---------------------------------------------------- |
+| `inbound_messages_size`  | `int64`    | Size of received messages in bytes since last metric. |
+| `outbound_messages_size` | `int64`    | Size of sent messages in bytes since last metric.     |
+| `serialization_time`     | `duration` | Time spent serializing messages since last metric.    |
+| `deserialization_time`   | `duration` | Time spent deserializing messages since last metric.  |
+
+Each record in the `actors` list has the following schema:
+
+| Field             | Type       | Description                                       |
+| :---------------- | :--------- | :------------------------------------------------ |
+| `name`            | `string`   | Name of the actor.                                |
+| `processing_time` | `duration` | Time spent processing messages since last metric. |
```

Comment:
Unfortunate that we don't get this by message handler.

Reply:
Yes, that would've been really nice to have. I asked about this upstream, and unfortunately there's no desire to add this or to even add actor IDs in addition to actor names to the dimensions.